### PR TITLE
Travis ice36

### PR DESCRIPTION
**Pull request by jburel**
In this PR:
* Use 16.04 image
* Use ice 3.6
* Change installed/upgraded version 

This PR currently uses the wheel for Ubuntu from gitlab.
This will need to be modified before this PR is merged

cc @manics

### CODE REVIEW

**Comment by sbesson** [comment]
:+1: do you want to split the `flake8` changes as these can probably be reviewed and merged immediately?

**Comment by jburel** [comment]
I can do but i don't think there is any rush to have that in.

File: travis-build [hunk 1]
```diff
@@ -11,26 +11,27 @@ pip install dist/*.tar.gz
...
 
   # Check db dump file
   omego db dump --serverdir OMERO.server --dumpfile travis-omero.pgdump
-  pg_restore -e travis-omero.pgdump | grep 'CREATE TABLE dbpatch'
+  
+  pg_restore -l travis-omero.pgdump | grep 'dbpatch_versions_trigger'
 fi
 
-#Test a multistage DB upgrade (4.4 -> 5.3) as part of the server upgrade
+#Test a multistage DB upgrade (5.3 -> 5.4) as part of the server upgrade
```
Comment (manics) [inline travis-build]: Could you make this 5.2 -> 5.4 instead (https://downloads.openmicroscopy.org/omero/5.2.8/ includes an ice 3.6 server)? The intention is to test an OMERO DB upgrade where there's not a direct upgrade path, e.g. to go from 5.2 to 5.4 you must run two files:
- `sql/psql/OMERO5.3__0/OMERO5.2__0.sql`
- `sql/psql/OMERO5.4__0/OMERO5.3__0.sql`

File: .travis.yml [hunk 2]
```diff
@@ -25,7 +26,18 @@ before_install:
...
+  - sudo apt-get -y install db5.3-util
+  - sudo apt-get -y install libssl-dev libbz2-dev libmcpp-dev libdb++-dev libdb-dev
+
+  - sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 5E6DA83306132997
+  - sudo apt-add-repository "deb http://zeroc.com/download/apt/ubuntu`lsb_release -rs` stable main"
+  - sudo apt-get update
+  - sudo apt-get -y install zeroc-ice-all-runtime zeroc-ice-all-dev
+  #- sudo pip install -U "zeroc-ice<3.7"
+  - sudo curl -L -O https://gitlab.com/openmicroscopy/incubator/zeroc-ice-py-ubuntu1604/-/jobs/129805187/artifacts/raw/dist/zeroc_ice-3.6.4-cp27-cp27mu-linux_x86_64.whl
+  - sudo pip install -U zeroc_ice-3.6.4-cp27-cp27mu-linux_x86_64.whl
```
Comment (manics) [inline .travis.yml]: FYI you can pip install a URL directly `pip install -U https://gitlab.com/openmicroscopy/incubator/zeroc-ice-py-ubuntu1604/-/jobs/129805187/artifacts/raw/dist/zeroc_ice-3.6.4-cp27-cp27mu-linux_x86_64.whl`

File: travis-build (hunk 1 above)
Comment (jburel) [reply travis-build]: I did not check that 
Thanks

File: .travis.yml (hunk 2 above)
Comment (jburel) [reply .travis.yml]: I used two lines mainly for clarity at that stage.
That will be replaced when we have decided of the final location

File: travis-build (hunk 1 above)
Comment (jburel) [reply travis-build]: and we will be going to 5.5 as part of this release

File: .travis.yml (hunk 2 above)
Comment (jburel) [reply .travis.yml]: I did a bit of investigation:
* 5.2 server will not start on Xenial travis docker image due to the Java version installed by default.
* An option is to keep the 14.04 image and use the wheel for 14.04 (we will need for https://github.com/openmicroscopy/openmicroscopy/)
* then move to 16.04 image when we have 5.5 out so we can do an upgrade from 5.3->5.5

What do you think?

File: travis-build (hunk 1 above)
Comment (jburel) [reply travis-build]: I did a bit of investigation:
* 5.2 server will not start on Xenial travis docker image due to the Java version installed by default.
* An option is to keep the 14.04 image and use the wheel for 14.04 (we will need for https://github.com/openmicroscopy/openmicroscopy/ not available yet)
* then move to 16.04 image when we have 5.5 out so we can do an upgrade from 5.3-> 5.5

What do you think?

Comment (manics) [reply travis-build]: Assuming no omego work is planned we could keep it as 5.3->5.4, and add an issue as a reminder to change it to 5.5?

Comment (jburel) [reply travis-build]: sounds good. So I will roll back the commits and when we have the wheel in its final location
update the pip installation

Comment (jburel) [reply travis-build]: https://github.com/ome/omego/issues/117

Comment (manics) [reply travis-build]: Thanks, this looks fine now

**Pull request merged**